### DISCUSSION TITLE
Invalid private or public key

### DISCUSSION THREAD

**Discussion by 0xAlon** [rednblkx/HomeKey-ESP32#36] [Q&A] [answered]
I’m encountering an error while using the Apple Home key. Has anyone else experienced this problem and know how to fix it?

> I (40783) LockMechanism: loop > *** SELECT HOMEKEY APPLET SUCCESSFUL ***
I (40922) HKAuthCtx: HKAuthenticationContext > Initialization Time: 139 ms
W (41123) HKFastAuth: attest > FAST Flow failed!
E (41124) CCUtils: signSharedInfo > ecp_read - ECP - Invalid private or public key
E (41282) HKStdAuth: attest > Response Status not 0x90, something went wrong!
W (41326) LockMechanism: loop > We got status FlowFailed, mqtt untouched!

**Comment by 50watt** [comment]
@0xAlon Same for me :-(

**Reply by naphanz** [reply]
me too

**Comment by rednblkx** [comment]
Hi there!

Please set the log level to verbose with `@LV` and let me know if “Reader Private Key” and TLV 4d[16] it’s all zeroes. This might be related to #34

**Reply by 5and0** [reply]
Also getting this and yes ... lots of zeros ... 

```
I (789644) LockMechanism: loop > *** SELECT HOMEKEY APPLET SUCCESSFUL ***
D (789649) LockMechanism: loop > Reader Private Key: 0000000000000000000000000000000000000000000000000000000000000000
```

`D` (789869) Utils::simple_tlv: TLV 4d[16]: 00000000000000000000000000000000

**Reply by 5and0** [reply]
I am also getting the following when paring ... /

`*** ERROR: Unrecognized Controller ID: `

**Reply by 5and0** [reply]
@rednblkx I have emailed you the logs I have captured. 
Thanks in advance

**Reply by 50watt** [reply]
Same as @5and0 for me:
`I (51216) LockMechanism: loop > *** SELECT HOMEKEY APPLET SUCCESSFUL ***`
`D (51221) LockMechanism: loop > Reader Private Key: 0000000000000000000000000000000000000000000000000000000000000000`
`....`
`D (51469) Utils::simple_tlv: TLV 4d[16]: 00000000000000000000000000000000`

**Reply by rednblkx** [reply]
What iOS version are you all running? 
17.4 got recently released and I haven’t updated to that, I’ll be checking that later when I arrive home but for the time being let’s see if this is a potential cause.

**Reply by 5and0** [reply]
Yes … 17.4 here

**Comment by rednblkx** [comment, marked as answer]
Think i have found the issue, it was my bad, please all pull/download the main branch and let me know if that fixes it, make sure to re-add the device into home.

**Reply by 0xAlon** [reply]
Yes, the issue has been resolved. Thanks!

**Reply by 5and0** [reply]
🙏 Thank you ! Working now !
